instrumentation/net/http/otelhttp/labeler.go

```diff
@@ -21,45 +21,57 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
-// Labeler is used to allow instrumented HTTP handlers to add custom attributes to
+type labelerContextKeyType int
+
+// LabelerContextKey is the key in the context.Context on which the Labeler instance would be placed.
+const LabelerContextKey labelerContextKeyType = 0
+
+// Labeler provides a way to hook custom attribute.KeyValue entries to a request context during the execution of the
+// request. The instance of the custom Labeler can be accessed from the request's context with the LabelerContextKey.
+type Labeler interface {
```

Comment:
This is a breaking change to this package's API.  If this can be accomplished without changing the `Labeler` type it can be considered.

Reply:
Argh, true. Let's keep the `Labeler` a struct. Can you suggest a fitting name for the interface?